subsys/math/cmsis_backend/public_overrides/dsp_backend/dsp.h

```diff
@@ -0,0 +1,99 @@
+/* Copyright (c) 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef SUBSYS_MATH_CMSIS_BACKEND_PUBLIC_OVERRIDES_DSP_BACKEND_TYPES_H_
+#define SUBSYS_MATH_CMSIS_BACKEND_PUBLIC_OVERRIDES_DSP_BACKEND_TYPES_H_
+
+#include <arm_math.h>
+
+typedef int8_t q1_7_t;
+typedef int16_t q1_15_t;
+typedef int32_t q1_31_t;
+typedef int32_t q9_23_t;
+typedef int32_t q16_16_t;
+typedef int32_t q24_8_t;
```

Comment:
> q1_7_t: 1 signed integer bit, 7 decimal precision bits
q1_15_t: 1 signed integer bit, 15 decimal precision bits
q1_31_t: 1 signed integer bit, 31 decimal precision bits

You are mapping these types to CMSIS-DSP `q7_t`, `q15_t` and `q31_t`. For CMSIS-DSP, these are all 0 integer bit (ARM Q notation includes sign bit in the "integer" part; would be Q0.31 in the TI notation), so its range is from -1 to +1.

> typedef int32_t q9_23_t;
typedef int32_t q16_16_t;
typedef int32_t q24_8_t;

This type system does not really make much sense to me.

No other math/DSP library I have seen does this, and they do not need to because there is no fundamental difference in the operations of, for instance, Q1.31 (`q31_t` in CMSIS-DSP) and the proposed Q9.23 (`q9_23_t`).

What matters is how the inputs and outputs are interpreted (scaled). You can feed Q9.23 to Q1.31 functions, as long as you interpret (scale) the outputs as such.

Moreover, if we decide to go down this path, there will be no end to this (e.g. why do we support `q9_23_t` and not `q10_22_t`, are we doomed to implement the functions for every single variant out there? -- that will be tens of thousands of functions).

`q7_t`, `q15_t`, `q31_t` for fixed point types should be enough.